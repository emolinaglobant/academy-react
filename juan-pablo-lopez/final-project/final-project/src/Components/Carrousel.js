import React from "react";
import carrousel1 from "../Img/carrousel1.jpg"
import carrousel2 from "../Img/carrousel2.jpg"
import carrousel3 from "../Img/carrousel3.jpg"

import {
  MDBCarousel,
  MDBCarouselInner,
  MDBCarouselItem,
  MDBCarouselElement,
  MDBCarouselCaption,
} from "mdb-react-ui-kit";

export const Carrousel = () => {
  return (
    <MDBCarousel id="carrousel">
      <MDBCarouselInner>
        <MDBCarouselItem className="active">
          <MDBCarouselElement
            src={carrousel1}
            alt="GIGABYTE MOTHERBOARDS"
          />
          <MDBCarouselCaption>
            <h5>GIGABYTE</h5>
            <p>GIGABYTE Ultra Durableâ„¢ motherboards bring together a unique combination of features and technologies that offer users the absolute power</p>
          </MDBCarouselCaption>
        </MDBCarouselItem>
        <MDBCarouselItem>
          <MDBCarouselElement
            src={carrousel2}
            alt="RAM"
          />
          <MDBCarouselCaption>
            <h5>RAM</h5>
            <p>Get the best deals on DDR4 SDRAM Computer RAM and find everything you'll need to improve your home office setup.</p>
          </MDBCarouselCaption>
        </MDBCarouselItem>
        <MDBCarouselItem>
          <MDBCarouselElement
            src={carrousel3}
            alt="..."
          />
          <MDBCarouselCaption>
            <h5>Second slide label</h5>
            <p>Lorem ipsum dolor sit amet, consectetur adipiscing elit.</p>
          </MDBCarouselCaption>
        </MDBCarouselItem>
      </MDBCarouselInner>
    </MDBCarousel>
  );
};
