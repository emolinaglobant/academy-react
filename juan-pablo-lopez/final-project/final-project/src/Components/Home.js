import React, { useState, useEffect } from 'react'
import { NavBar } from './NavBar'
import { FooterPage } from './FooterPage'
import { Products } from './Products'
import { auth, fs } from '../Config/FirebaseConfig'
import { IndividualFilteredProduct } from './IndividualFilteredProduct'
import { TiArrowBack } from 'react-icons/ti'

export const Home = (props) => {

    let Product;
    const uid = GetUserUid();
    const user = GetCurrentUser();
    const [products, setProducts] = useState([]);
    const [totalProducts, setTotalProducts] = useState(0);
    const [active, setActive] = useState('');
    const [category, setCategory] = useState('');
    const [filteredProducts, setFilteredProducts] = useState([]);

    const addToCart = (product) => {
        if (uid !== null) {
            Product = product;
            Product['qty'] = 1;
            Product['TotalProductPrice'] = Product.qty * Product.price;
            fs.collection('Cart ' + uid).doc(product.ID).set(Product).then(() => {
                console.log('successfully added to cart');
            })
        }
        else {
            props.history.push('/login');
        }
    }

    function GetCurrentUser() {
        const [user, setUser] = useState(null);
        useEffect(() => {
            auth.onAuthStateChanged(user => {
                if (user) {
                    fs.collection('users').doc(user.uid).get().then(snapshot => {
                        setUser(snapshot.data().FullName);
                    })
                }
                else {
                    setUser(null);
                }
            })
        }, [])
        return user;
    }

    const getProducts = async () => {
        const products = await fs.collection('Products').get();
        const productsArray = [];
        for (var snap of products.docs) {
            var data = snap.data();
            data.ID = snap.id;
            productsArray.push({
                ...data
            })
            if (productsArray.length === products.docs.length) {
                setProducts(productsArray);
            }
        }
    }

    function GetUserUid() {
        const [uid, setUid] = useState(null);
        useEffect(() => {
            auth.onAuthStateChanged(user => {
                if (user) {
                    setUid(user.uid);
                }
            })
        }, [])
        return uid;
    }

    useEffect(() => {
        getProducts();
    }, [])

    useEffect(() => {
        auth.onAuthStateChanged(user => {
            if (user) {
                fs.collection('Cart ' + user.uid).onSnapshot(snapshot => {
                    const qty = snapshot.docs.length;
                    setTotalProducts(qty);
                })
            }
        })
    }, [])


    const [spans] = useState([
        { id: 'GraphicsCards', text: 'Graphics Cards' },
        { id: 'Keyboards', text: 'Keyboards' },
        { id: 'ComputerCases', text: 'Computer Cases' },
        { id: 'Monitors', text: 'Monitors' },
        { id: 'Motherboards', text: 'Motherboards' },
        { id: 'Laptops', text: 'Laptops' },
        { id: 'StorageUnits', text: `Storage units` },
        { id: 'RamMemory', text: `RAM Memory` },
        { id: 'FontPower', text: 'Font power' },
    ])

    const handleChange = (individualSpan) => {
        setActive(individualSpan.id);
        setCategory(individualSpan.text);
        filterFunction(individualSpan.text);
    }

    const filterFunction = (text) => {
        if (products.length > 1) {
            const filter = products.filter((product) => product.category === text);
            setFilteredProducts(filter);
        }
        else {
            console.log('no products to filter')
        }
    }

    const returntoAllProducts = () => {
        setActive('');
        setCategory('');
        setFilteredProducts([]);
    }

    return (
        <>
            <title>learn react</title>
            <NavBar user={user} totalProducts={totalProducts} />
            <br></br>
            <main className='container-fluid filter-products-main-box'>
                <section className='filter-box' id="filter">
                    <h6>Filter by category</h6>
                    {spans.map((individualSpan, index) => (
                        <span key={index} id={individualSpan.id}
                            onClick={() => handleChange(individualSpan)}
                            className={individualSpan.id === active ? active : 'deactive'}>{individualSpan.text}</span>
                    ))}
                </section>
                {filteredProducts.length > 0 && (
                    <section className='my-products' id="filtered-product">
                        <h1 className='text-center'>{category}</h1>
                        <a href="/" onClick={returntoAllProducts}><TiArrowBack size={35} />Return to All Products</a>
                        <section className='products-box'>
                            {filteredProducts.map(individualFilteredProduct => (
                                <IndividualFilteredProduct key={individualFilteredProduct.ID}
                                    individualFilteredProduct={individualFilteredProduct}
                                    addToCart={addToCart} />
                            ))}
                        </section>
                    </section>
                )}
                {filteredProducts.length < 1 && (
                    <>
                        {products.length > 0 && (
                            <section className='my-products'>
                                <h1 className='text-center'>All Products</h1>
                                <section className='products-box'>
                                    <Products products={products} addToCart={addToCart} />
                                </section>
                            </section>
                        )}
                        {products.length < 1 && (
                            <section className='my-products please-wait'>Please wait...</section>
                        )}
                    </>
                )}
            </main>
            <FooterPage></FooterPage>
        </>
    )
}