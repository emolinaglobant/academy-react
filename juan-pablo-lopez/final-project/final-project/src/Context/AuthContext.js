import { createContext, useContext, useEffect, useState } from "react";
import {
  onAuthStateChanged,
  GoogleAuthProvider,
  signInWithPopup,
  sendPasswordResetEmail,
} from "firebase/auth";
import { auth } from '../Config/FirebaseConfig';

const authContext = createContext();

export const useAuth = () => {
  const context = useContext(authContext);
  if (!context) throw new Error("There is no Auth provider");
  return context;
};

export function AuthProvider({ children }) {
  const [userData, setUserData] = useState(null);
  const [loading, setLoading] = useState(true);

  const loginWithGoogle = () => {
    const googleProvider = new GoogleAuthProvider();
    return signInWithPopup(auth, googleProvider);
  };

  const resetPassword = async (email) => sendPasswordResetEmail(auth, email);

  useEffect(() => {
    const unsubuscribe = onAuthStateChanged(auth, (currentUser) => {
      console.log({ currentUser });
      setUserData(currentUser);
      setLoading(false);
    });
    return () => unsubuscribe();
  }, []);

  return (
    <authContext.Provider
      value={{
        userData,
        loading,
        loginWithGoogle,
        resetPassword,
      }}
    >
      {children}
    </authContext.Provider>
  );
}