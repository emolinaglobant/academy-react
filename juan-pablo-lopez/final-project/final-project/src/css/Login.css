.login,
.image {
    min-height: 100vh;
}
.bg-image {
    background-image: url("../Img/login.gif");
    background-size: cover;
    background-position: center center;
}

.link-span{
    text-align: center;
    padding: 20px 20px;
}

.google-btn {
  width: 184px;
  height: 42px;
  background-color: #4285f4;
  border-radius: 2px;
  box-shadow: 0 3px 4px 0 rgba(0,0,0,.25);
  margin-left: 125px;
}

  .google-icon-wrapper {
    position: absolute;
    margin-top: 1px;
    margin-left: 1px;
    width: 40px;
    height: 40px;
    border-radius: 2px;
    background-color: #fff;;
  }
  .google-icon {
    position: absolute;
    margin-top: 11px;
    margin-left: 11px;
    width: 18px;
    height: 18px;
  }
  .btn-text {
    float: right;
    margin: 11px 11px 0 0;
    color: #fff;;
    font-size: 14px;
    letter-spacing: 0.2px;
    font-family: "Roboto";
  }
  .btn-text:hover {
    box-shadow: 0 0 6px #1669F2;
  }

  .btn-text:active {
    background: #1669F2;
  }


  .btn-box {
    display: flex;
    justify-content: center;
}

a {
    text-decoration: none;
}

.container_login{
    margin: 150px;

}
@import url(https://fonts.googleapis.com/css?family=Roboto:500);