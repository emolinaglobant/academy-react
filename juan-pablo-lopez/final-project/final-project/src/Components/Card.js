import React from "react";
import PropTypes from "prop-types";
import "../css/Card.css";

export const Card = ({ imageSource, title, text, url }) => {
  return (
    <>
      <main className="card text-center bg-dark animate__animated animate__fadeInUp">
        <section className="overflow">
          <img src={imageSource} alt="a wallpaper" className="card-img-top" />
        </section>
        <section className="card-body text-light">
          <h4 className="card-title">{title}</h4>
          <p className="card-text text-secondary">
            {text
              ? text
              : "Show More"}
          </p>
          <a
            href={url ? url : ""}
            target="_blank"
            className="btn btn-outline-secondary border-0"
            rel="noreferrer"
          >
            Go to {title}
          </a>
        </section>
      </main>
    </>
  );
}

Card.propTypes = {
  title: PropTypes.string.isRequired,
  text: PropTypes.string,
  url: PropTypes.string,
  imageSource: PropTypes.string
};

export default Card;