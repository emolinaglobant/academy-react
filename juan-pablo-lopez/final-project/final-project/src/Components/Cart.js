import React, { useState, useEffect } from 'react'
import { NavBar } from './NavBar'
import { auth, fs } from '../Config/FirebaseConfig'
import { CartProducts } from './CartProducts';
import StripeCheckout from 'react-stripe-checkout';
import { toast } from 'react-toastify';
import axios from 'axios';
import { useHistory } from 'react-router-dom'
import 'react-toastify/dist/ReactToastify.css';

toast.configure();

export const Cart = () => {
    const user = GetCurrentUser();
    const [cartProducts, setCartProducts] = useState([]);

    function GetCurrentUser() {
        const [user, setUser] = useState(null);
        useEffect(() => {
            auth.onAuthStateChanged(user => {
                if (user) {
                    fs.collection('users').doc(user.uid).get().then(snapshot => {
                        setUser(snapshot.data().FullName);
                    })
                }
                else {
                    setUser(null);
                }
            })
        }, [])
        return user;
    }
    
    useEffect(() => {
        auth.onAuthStateChanged(user => {
            if (user) {
                fs.collection('Cart ' + user.uid).onSnapshot(snapshot => {
                    const newCartProduct = snapshot.docs.map((doc) => ({
                        ID: doc.id,
                        ...doc.data(),
                    }));
                    setCartProducts(newCartProduct);
                })
            }
            else {
                console.log('user is not signed in to retrieve cart');
            }
        })
    }, [])

    const qty = cartProducts.map(cartProduct => {
        return cartProduct.qty;
    })

    const reducerOfQty = (accumulator, currentValue) => accumulator + currentValue;
    const totalQty = qty.reduce(reducerOfQty, 0);
    const price = cartProducts.map((cartProduct) => {
        return cartProduct.TotalProductPrice;
    })
    
    const reducerOfPrice = (accumulator, currentValue) => accumulator + currentValue;
    const totalPrice = price.reduce(reducerOfPrice, 0);
    let Product;

    
    const cartProductIncrease = (cartProduct) => {
        Product = cartProduct;
        Product.qty = Product.qty + 1;
        Product.TotalProductPrice = Product.qty * Product.price;
        auth.onAuthStateChanged(user => {
            if (user) {
                fs.collection('Cart ' + user.uid).doc(cartProduct.ID).update(Product).then(() => {
                    console.log('increment added');
                })
            }
            else {
                console.log('user is not logged in to increment');
            }
        })
    }

    const cartProductDecrease = (cartProduct) => {
        Product = cartProduct;
        if (Product.qty > 1) {
            Product.qty = Product.qty - 1;
            Product.TotalProductPrice = Product.qty * Product.price;
            auth.onAuthStateChanged(user => {
                if (user) {
                    fs.collection('Cart ' + user.uid).doc(cartProduct.ID).update(Product).then(() => {
                        console.log('decrement');
                    })
                }
                else {
                    console.log('user is not logged in to decrement');
                }
            })
        }
    }

    const [totalProducts, setTotalProducts] = useState(0); 
    useEffect(() => {
        auth.onAuthStateChanged(user => {
            if (user) {
                fs.collection('Cart ' + user.uid).onSnapshot(snapshot => {
                    const qty = snapshot.docs.length;
                    setTotalProducts(qty);
                })
            }
        })
    }, [])

    const history = useHistory();
    const handleToken = async (token) => {
        const cart = { name: 'All Products', totalPrice }
        const response = await axios.post('http://localhost:8080/checkout', {
            token,
            cart
        })
        console.log(response);
        let { status } = response.data;
        console.log(status);
        if (status === 'success') {
            history.push('/');
            toast.success('Your order has been placed successfully', {
                position: 'top-right',
                autoClose: 5000,
                hideProgressBar: false,
                closeOnClick: true,
                pauseOnHover: false,
                draggable: false,
                progress: undefined,
            });

            const uid = auth.currentUser.uid;
            const carts = await fs.collection('Cart ' + uid).get();
            for (var snap of carts.docs) {
                fs.collection('Cart ' + uid).doc(snap.id).delete();
            }
        }
        else {
            alert('Something went wrong in checkout');
        }
    }

    return (
        <>
            <NavBar user={user} totalProducts={totalProducts} />
            <br></br>
            {cartProducts.length > 0 && (
                <main className='container-fluid'>
                    <h1 className='text-center'>Cart</h1>
                    <section className='products-box'>
                        <CartProducts cartProducts={cartProducts}
                            cartProductIncrease={cartProductIncrease}
                            cartProductDecrease={cartProductDecrease}
                        />
                    </section>
                    <section className='summary-box'>
                        <h5>Cart Summary</h5>
                        <br></br>
                        <section>
                            Total No of Products: <span>{totalQty}</span>
                        </section>
                        <section>
                            Total Price to Pay: <span>$ {totalPrice}</span>
                        </section>
                        <br></br>
                        <StripeCheckout
                            stripeKey='pk_test_51KW5K8BUzjitVX7lRP6rCHB2JwfbqvxdNUaum3liStXdsPODPbBzo9WdiovCA13vXSXddYuRLVwzGKNH8VxmsofP00aw4NQskj'
                            token={handleToken}
                            billingAddress
                            shippingAddress
                            name='All Products'
                            amount={totalPrice * 100}
                        ></StripeCheckout>
                    </section>
                </main>
            )}
            {cartProducts.length < 1 && (
                <section className='container-fluid'>No products to show</section>
            )}
        </>
    )
}