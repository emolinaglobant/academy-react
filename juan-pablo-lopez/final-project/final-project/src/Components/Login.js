import React, { useState } from 'react'
import 'bootstrap/dist/css/bootstrap.min.css';
import 'bootstrap/dist/js/bootstrap.min.js';
import { Link } from 'react-router-dom'
import { auth } from '../Config/FirebaseConfig'
import { useHistory } from 'react-router-dom'
import { NavBar } from './NavBar'
import { useAuth } from '../Context/AuthContext'
import googleLogo from '../Img/google.svg'

export const Login = () => {

    const history = useHistory();
    const [email, setEmail] = useState('');
    const [password, setPassword] = useState('');
    const [errorMsg, setErrorMsg] = useState('');
    const [successMsg, setSuccessMsg] = useState('');
    const { loginWithGoogle, resetPassword } = useAuth();

    const handleLogin = (e) => {
        e.preventDefault();
        auth.signInWithEmailAndPassword(email, password).then(() => {
            setSuccessMsg('Login succesfull. You will now get automatically redirected to Home');
            setEmail('');
            setPassword('');
            setErrorMsg('');
            setTimeout(() => {
                setSuccessMsg('');
                history.push('/categories');
            }, 3000)
        }).catch(error => setErrorMsg(error.message));
    }

    const handleGoogleSignin = async () => {
        try {
            await loginWithGoogle();
            history.push('/categories');
        } catch (error) {
            setErrorMsg(error.message);
        }
    };

    const handleResetPassword = async (e) => {
        e.preventDefault();
        if (!email) return setErrorMsg("Write an correct email to reset password");
        try {
            await resetPassword(email);
            setSuccessMsg('We sent you an email. Check your inbox')
        } catch (error) {
            setErrorMsg(error.message);
        }
    };

    return (
        <>
            <NavBar></NavBar>
            <main className="maincontainer">
                <section className="container-fluid">
                    <section className="row no-gutter">
                        <section className="col-md-6 d-none d-md-flex bg-image"></section>
                        <section className="col-md-6 bg-light">
                            <section className="login d-flex align-items-center py-5">
                                <section className="container">
                                    <section className="row">
                                        <section className="col-lg-10 col-xl-7 mx-auto">
                                            <h3 className="display-4">Login</h3>
                                            {successMsg && <>
                                                <section className='success-msg'>{successMsg}</section>
                                                <br></br>
                                            </>}
                                            {errorMsg && <>
                                                <section className='error-msg'>{errorMsg}</section>
                                                <br></br>
                                            </>}
                                            <br></br>
                                            <form autoComplete="off" onSubmit={handleLogin}>
                                                <section className="mb-3" >
                                                    <label>Email*</label>
                                                    <input id="inputEmail" type="email" required placeholder="Email address" onChange={(e) => setEmail(e.target.value)} value={email} autoFocus="" className="form-control rounded-pill border-0 shadow-sm px-4" />
                                                </section>
                                                <label>Password*</label>
                                                <section className="mb-3">
                                                    <input id="inputPassword" type="password" placeholder="Password" required="" onChange={(e) => setPassword(e.target.value)} value={password} className="form-control rounded-pill border-0 shadow-sm px-4 text-primary" />
                                                </section>
                                                <section className="form-check">
                                                    <input id="customCheck1" type="checkbox" checked className="form-check-input" />
                                                    <label htmlFor="customCheck1" className="form-check-label">Remember password</label>
                                                </section>
                                                <section className="d-grid gap-2 mt-2">
                                                    <button type="submit" className="btn btn-primary btn-block text-uppercase mb-2 rounded-pill shadow-sm">LOGIN</button>
                                                    <section className="btn-box">
                                                        <a className="inline-block align-baseline font-bold text-sm text-blue-500 hover:text-blue-800" href="#!" onClick={handleResetPassword}>Forgot Password?</a>
                                                        <span><Link to="recover" className="link"></Link></span>
                                                    </section>
                                                </section>
                                                <section className="btn-box">
                                                    <span className="link-span">Don't have an account? Signup
                                                        <Link to="signup" className="link"> Here</Link></span>

                                                </section>
                                            </form>
                                            <section className="google-btn" onClick={handleGoogleSignin}>
                                                <section className="google-icon-wrapper">
                                                    <img className="google-icon" src={googleLogo} alt="Google Logo" />
                                                </section>
                                                <p className="btn-text"><b>Sign in with google</b></p>
                                            </section>
                                        </section>
                                    </section>
                                </section>
                            </section>
                        </section>
                    </section>
                </section>
            </main>
        </>
    )
};

