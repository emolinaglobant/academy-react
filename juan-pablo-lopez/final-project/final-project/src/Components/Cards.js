import React from "react";
import Card from "./Card";
import { NavBar } from './NavBar'
import {Carrousel} from './Carrousel'
import { FooterPage } from "./FooterPage";
import image1 from '../Img/graphic.gif'
import image2 from '../Img/keyboard1.gif'
import image3 from '../Img/case.gif'
import image4 from '../Img/monitor.gif'
import image5 from '../Img/motherboard.gif'
import image6 from '../Img/laptop.gif'
import image7 from '../Img/storage.gif'
import image8 from '../Img/ram.gif'
import image9 from '../Img/fontpower.gif'


const cards = [
  {
    id: 1,
    title: "Graphics Cards",
    image: image1,
    url: "categories",
  },
  {
    id: 2,
    title: "Keyboards",
    image: image2,
    url: "categories",
  },
  {
    id: 3,
    title: "Computer Cases",
    image: image3,
    url: "categories",
  },
  {
    id: 4,
    title: "Monitors",
    image: image4,
    url: "categories",
  },
  {
    id: 5,
    title: "Motherboards",
    image: image5,
    url: "categories",
  },
  {
    id: 6,
    title: "Laptops",
    image: image6,
    url: "categories",
  },
  {
    id: 7,
    title: "Storage Units",
    image: image7,
    url: "categories",
  },
  {
    id: 8,
    title: "RAM",
    image: image8,
    url: "categories",
  },
  {
    id: 9,
    title: "Font Power",
    image: image9,
    url: "categories",
  },
];

export const Cards = () => {
  return (
    <>
    <NavBar ></NavBar>
    <Carrousel></Carrousel>
    <br></br>
    <main>
    <section className="container d-flex justify-content-center align-items-center h-100">
      <section className="row">
        {cards.map(({ title, image, url, id }) => (
          <section className="col-md-4" key={id} id="cards">
            <Card imageSource={image} title={title} url="/categories" /><br></br>
          </section>
        ))}
      </section>
    </section>
    </main>
    <FooterPage></FooterPage>
    </>
  );
  
}

export default Cards;