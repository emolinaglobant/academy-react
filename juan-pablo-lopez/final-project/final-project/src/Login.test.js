import React from 'react';
import { shallow } from 'enzyme';
import { Login } from './Components/Login'
import { signInWithEmailAndPassword } from './Context/AuthContext'
import {isAuthenticated} from "firebase/app";
import { Home } from './Components/Home'

// it('should render span content', () => {
//     const wrapper = shallow(<Login/>);
//     const span = wrapper.find('span');
//     const result = span.text();

//     expect(result).toBe("Don't have an account? Signup");
// });

// test('signInWithEmailAndPassword should throw error with wrong credential', async () => {
//     let error = '';
//     try {
//         await signInWithEmailAndPassword('unit-testing@testing.com', "a");
    
//     } catch(err) {
//         error = err.toString();
//     }

//     expect(error).toEquals(
//         'Error: the password is invalid or the user does not have a password'
//     );
// });

// test('signOutFirebase should work', async () => {
//     await signInWithEmailAndPassword('admin@gmail.com', 'admin123');
//     expect(isAuthenticated()).toBe(true);

// });
it('should render correctly with no props', () => {
    const component = shallow(<Home/>);
    
    expect(component).toMatchSnapshot();
  });