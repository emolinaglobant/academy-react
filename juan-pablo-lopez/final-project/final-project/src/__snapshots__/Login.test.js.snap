// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly with no props 1`] = `
<Fragment>
  <title>
    learn react
  </title>
  <NavBar
    totalProducts={0}
    user={null}
  />
  <br />
  <main
    className="container-fluid filter-products-main-box"
  >
    <section
      className="filter-box"
      id="filter"
    >
      <h6>
        Filter by category
      </h6>
      <span
        className="deactive"
        id="GraphicsCards"
        key="0"
        onClick={[Function]}
      >
        Graphics Cards
      </span>
      <span
        className="deactive"
        id="Keyboards"
        key="1"
        onClick={[Function]}
      >
        Keyboards
      </span>
      <span
        className="deactive"
        id="ComputerCases"
        key="2"
        onClick={[Function]}
      >
        Computer Cases
      </span>
      <span
        className="deactive"
        id="Monitors"
        key="3"
        onClick={[Function]}
      >
        Monitors
      </span>
      <span
        className="deactive"
        id="Motherboards"
        key="4"
        onClick={[Function]}
      >
        Motherboards
      </span>
      <span
        className="deactive"
        id="Laptops"
        key="5"
        onClick={[Function]}
      >
        Laptops
      </span>
      <span
        className="deactive"
        id="StorageUnits"
        key="6"
        onClick={[Function]}
      >
        Storage units
      </span>
      <span
        className="deactive"
        id="RamMemory"
        key="7"
        onClick={[Function]}
      >
        RAM Memory
      </span>
      <span
        className="deactive"
        id="FontPower"
        key="8"
        onClick={[Function]}
      >
        Font power
      </span>
    </section>
    <section
      className="my-products please-wait"
    >
      Please wait...
    </section>
  </main>
  <FooterPage />
</Fragment>
`;
